kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/filter/DenyCipherSuiteFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import io.netty.handler.ssl.CipherSuiteFilter;
+import io.netty.util.internal.EmptyArrays;
+
+/**
+ * This class has been built to filter cipher suites based on requirements for Kroxylicious
+ * If no ciphers are declared then the default platform ciphers will be used
+ * If ciphers are declared then they are used instead of the default platform ciphers
+ * Both lists would then have anything removed that wasn't supported or in the deniedCiphers
+ */
+public final class DenyCipherSuiteFilter implements CipherSuiteFilter {
+
+    public static final DenyCipherSuiteFilter INSTANCE = new DenyCipherSuiteFilter();
+
+    private DenyCipherSuiteFilter() {
+    }
+
+    private List<String> deniedCiphers;
+
+    public DenyCipherSuiteFilter deniedCiphers(List<String> deniedCiphers) {
+        this.deniedCiphers = deniedCiphers == null ? new ArrayList<>() : deniedCiphers;
+        return this;
+    }
+
+    @Override
+    public String[] filterCipherSuites(Iterable<String> ciphers, List<String> defaultCiphers,
+                                       Set<String> supportedCiphers) {
+        return ciphers == null ? allowedCiphers(defaultCiphers.size(), defaultCiphers, supportedCiphers)
```

Comment:
you can probably apply some Java streams goodness(TM).  

```java
        var actualCiphers = (ciphers == null || !ciphers.iterator().hasNext()) ? defaultCiphers : ciphers;

        return StreamSupport.stream(actualCiphers.spliterator(), true)
                .filter(supportedCiphers::contains)
                .filter(Predicate.not(c -> deniedCiphers.contains(c)))
                .toList()
                .toArray(new String[] {});
```

Reply:
Also I think we should log (warn) if either ciphers or deniedCiphers contain a cipher suite name that's not  in the supportedCiphers set.   It could be that a user has typo'd the name of teh cipher suite they are trying to exclude.

Reply:
Yes, I did have some logging in when doing the development actually, but stripped it after finishing. Will add in the warn for things that may be wrong.

I can definitely add some streams goodness haha. As mentioned above I'd taken inspiration from the Netty Identity* and Supported* implemented classes which suited my old dog/no new tricks coding!

Reply:
The code you have above, could the parallel true potentially risk the order of the list passed in? I believe the ciphers are in order of preference

Reply:
I doubt we should be enabling parallel streams in this case, the lists are going to be so short the costs of going parallel are not worth it.